### ISSUE TITLE
Inline Admonition Overlap

### ISSUE THREAD

**Issue by jacobswe** [squidfunk/mkdocs-material#2417]
- [x] I've read the [contribution guidelines][1] and agree with them

__I've found a bug and checked that ...__

- [x] ... the problem doesn't occur with the default MkDocs template
- [x] ... the problem is not in any of my customizations (CSS, JS, template)
- [x] ... the documentation does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem

## Description

Inline end admonitions are overlapped by regular admonitions when previously they would be well padded and contained.

### Expected behavior

No overlap between admonitions.

### Actual behavior

![image](https://user-images.githubusercontent.com/14115132/110357175-001e9f80-8009-11eb-9bb5-3e42d6170adb.png)

### Steps to reproduce the bug

Markdown:

```markdown
# Admonition Test

!!! info inline end 
    I too am an admonition.
    
!!! warning
    I am an admonition.
```

### Package versions

* Python: 3.7.3
* MkDocs: 1.1.2
* Material: 7.0.5+insiders.2.2.1

### Project configuration

``` yaml
site_name: My Docs
theme:
  name: material
markdown_extensions:
  - admonition
```

### System information

* OS: linux-gnu
* Browser: Chrome

  [1]: https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md

**Comment by squidfunk** [comment]
Thanks for reporting! This was caused by some changes to the CSS that were necessary to allow code block annotations inside of Admonitions. It should be fixed with the latest commit on Insiders - could you check the latest `master`?

**Comment by jacobswe** [comment]
Thank you for this shockingly fast fix! All looks good on my end at this newest commit. I'll go ahead and consider this issue closed then.

**Comment by polarathene** [comment]
Looks like the issue was flipped? This is against the `latest` Docker image with a fresh `docker pull`:

    !!! info inline end "Result:"

        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.

    **Markdown:**

    ```md
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.
    ```

![Screenshot_20210314_160309](https://user-images.githubusercontent.com/5098581/111055866-e7b5f580-84de-11eb-8e8b-c5f219f8e8c0.png)

---

    !!! info inline "Result:"

        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.

    **Markdown:**

    ```md
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.
    ```

![Screenshot_20210314_160539](https://user-images.githubusercontent.com/5098581/111055892-251a8300-84df-11eb-9eb6-d8cf2ce418d0.png)

---

## Related - Declaration order (Probably a documentation issue):

    **Markdown:**

    ```md
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.
    ```

    !!! info inline "Result:"

        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.

![Screenshot_20210314_160654](https://user-images.githubusercontent.com/5098581/111055902-509d6d80-84df-11eb-87e5-d4b6a0ba05d0.png)

---

    **Markdown:**

    ```md
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.
    ```

    !!! info inline end "Result:"

        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.

![Screenshot_20210314_160738](https://user-images.githubusercontent.com/5098581/111055913-7aef2b00-84df-11eb-92cc-ef7f78bb513c.png)

---

The [admonition inline docs](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#inline-blocks) should probably clarify that `inline`usage  should be declared prior to the block content that you want to inline the admonition with, otherwise it starts a new block, regardless of the width of the sibling content:

    **Markdown:**

    ```md
    Lorem ipsum dolor
    ```

    !!! info inline end "Result:"

        Lorem ipsum dolor

![Screenshot_20210314_161114](https://user-images.githubusercontent.com/5098581/111055965-08cb1600-84e0-11eb-9b71-a95346050de4.png)

    **Markdown:**

    !!! info inline end "Result:"

        Lorem ipsum dolor

![Screenshot_20210314_161250](https://user-images.githubusercontent.com/5098581/111055982-2ac49880-84e0-11eb-9760-2453393fb05e.png)

**Comment by squidfunk** [comment]
From what I can see, everything is working as intended.

**Comment by polarathene** [comment]
> From what I can see

Are you sure? Look at the 2nd image I posted, it's the same issue as depicted earlier here, there is no margin/gap between the two sections. Unlike the image prior to it which does add such between the code block and the admonition.

**Comment by squidfunk** [comment]
Ah right, missed that one, sorry! I'll look into it.

**Comment by polarathene** [comment]
I'm not familiar with how the community edition and insiders are maintained differently.

I was under the impression that Insiders feature gated it's exclusive features but carried over any bug fixes to community. Since this feature is available on the community edition, it seemed like that would have been pushed to the Docker image release too. All good :+1:

**Comment by squidfunk** [comment]
> I was under the impression that Insiders feature gated it's exclusive features but carried over any bug fixes to community. Since this feature is available on the community edition, it seemed like that would have been pushed to the Docker image release too. All good 👍

That's correct. However, the original reported error was not present in the community edition. Sometimes changes to adjacency modules/rules are necessary (in this case admonitions) that carry no value to be merged alone, as they don't fix anything.

**Comment by squidfunk** [comment]
FYI, I'm only able to reproduce this error if you use inline admonitions _inside admonitions_. If you use them on the top-level the margins are correct. This is not related to the current issue, but you can open a new one if you like:

<img width="723" alt="Bildschirmfoto 2021-03-14 um 12 55 29" src="https://user-images.githubusercontent.com/932156/111067446-94cf4480-84c4-11eb-8f5b-c5797f28bf5f.png">

Markdown to reproduce:

````markdown
!!! example "Example"

    !!! info inline "Result:"

        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.

    **Markdown:**

    ```md
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.
    ```

---

!!! info inline end "Result:"

    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.

**Markdown:**

```md
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.
```

---

!!! info inline "Result:"

    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.

**Markdown:**

```md
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.
```
````

**Comment by squidfunk** [comment]
That was a fast fix – c6e15970.
